src/Overlay.tsx

```diff
@@ -0,0 +1,123 @@
+import styled from 'styled-components'
+import React, {ReactElement} from 'react'
+import {get, COMMON, POSITION, SystemPositionProps, SystemCommonProps} from './constants'
+import {ComponentProps} from './utils/types'
+import {useOverlay, AnchoredPositionHookSettings, TouchOrMouseEvent} from './hooks'
+import Portal from './Portal'
+import sx, {SxProp} from './sx'
+
+type StyledOverlayProps = {
+  width?: keyof typeof widthMap
+  height?: keyof typeof heightMap
+  visibility: 'visible' | 'hidden'
+}
+
+const heightMap = {
+  sm: '480px',
+  md: '640px',
+  auto: 'auto'
+}
+
+const widthMap = {
+  sm: '256px',
+  md: '320px',
+  lg: '480px',
+  xl: '640px',
+  auto: 'auto'
+}
+
+/*TODO replace with shadow functional color variable when it's shipped to primer/primitives*/
+const StyledOverlay = styled.div<StyledOverlayProps & SystemCommonProps & SystemPositionProps & SxProp>`
+  background-color: ${get('colors.bg.overlay')};
+  box-shadow: ${get('overlay.boxShadow')};
+  position: absolute;
+  min-width: 192px;
+  max-width: 640px;
+  height: ${props => heightMap[props.height || 'auto']};
+  width: ${props => widthMap[props.width || 'auto']};
+  border-radius: ${get('overlay.borderRadius')};
+  overflow: hidden;
+  animation: overlay-appear 200ms ${get('animation.easeOutCubic')};
+
+  @keyframes overlay-appear {
+    0% {
+      opacity: 0;
+      transform: translateY(${get('space.2')});
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+  visibility: ${props => props.visibility};
+  ${COMMON};
+  ${POSITION};
+  ${sx};
+`
+export type OverlayProps = {
+  ignoreClickRefs: React.RefObject<HTMLElement>[]
+  initialFocusRef?: React.RefObject<HTMLElement>
+  returnFocusRef: React.RefObject<HTMLElement>
+  anchorRef: React.RefObject<HTMLElement>
+  onClickOutside: (e: TouchOrMouseEvent) => void
+  onEscape: (e: KeyboardEvent) => void
+  positionSettings?: AnchoredPositionHookSettings
+  positionDeps?: React.DependencyList
+} & Omit<ComponentProps<typeof StyledOverlay>, 'visibility' | keyof SystemPositionProps>
+
+/**
+ * An `Overlay` is a flexible floating surface, used to display transient content such as menus,
+ * selection options, dialogs, and more. Overlays use shadows to express elevation. The `Overlay`
+ * component handles all behaviors needed by overlay UIs as well as the common styles that all overlays * should have.
+ * @param positionSettings Settings for calculating the anchored position.
```

Comment:
Prop docs below (e.g. `ignoreClickRefs` and `anchorRef`) are explicitly marked `Optional`/`Required`. Can we add that to `positionSettings` (and `positionDeps`, `width`, and `height`)?

Reply:
I suggested some changes on @T-Hugs PR #1132 to address this